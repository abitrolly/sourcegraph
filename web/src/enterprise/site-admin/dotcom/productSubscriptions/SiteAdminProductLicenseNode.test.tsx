import React from 'react'
import * as GQL from '../../../../../../shared/src/graphql/schema'
import { SiteAdminProductLicenseNode } from './SiteAdminProductLicenseNode'
import { MemoryRouter } from 'react-router'
import { mount } from 'enzyme'

jest.mock('../../../dotcom/productSubscriptions/AccountName', () => ({ AccountName: 'AccountName' }))

describe('SiteAdminProductLicenseNode', () => {
    test('active', () => {
        expect(
            mount(
                <MemoryRouter>
                    <SiteAdminProductLicenseNode
                        node={
                            {
                                createdAt: '2020-01-01',
                                id: 'l1',
                                licenseKey: 'lk1',
                                info: {
                                    __typename: 'ProductLicenseInfo',
                                    expiresAt: '2021-01-01',
                                    productNameWithBrand: 'NB',
                                    tags: ['a'],
                                    userCount: 123,
                                },
                                subscription: {
                                    name: 's',
                                    activeLicense: { id: 'l1' },
                                    urlForSiteAdmin: '/s',
                                } as GQL.IProductSubscription,
                            } as GQL.IProductLicense
                        }
                        showSubscription={true}
                    />
                </MemoryRouter>
            )
        ).toMatchSnapshot()
    })

    test('inactive', () => {
        expect(
            mount(
                <MemoryRouter>
                    <SiteAdminProductLicenseNode
                        node={
                            {
                                createdAt: '2020-01-01',
                                id: 'l1',
                                licenseKey: 'lk1',
                                info: {
                                    __typename: 'ProductLicenseInfo',
                                    expiresAt: '2021-01-01',
                                    productNameWithBrand: 'NB',
                                    tags: ['a'],
                                    userCount: 123,
                                },
                                subscription: {
                                    name: 's',
                                    activeLicense: { id: 'l0' },
                                    urlForSiteAdmin: '/s',
                                } as GQL.IProductSubscription,
                            } as GQL.IProductLicense
                        }
                        showSubscription={true}
                    />
                </MemoryRouter>
            )
        ).toMatchSnapshot()
    })
})
