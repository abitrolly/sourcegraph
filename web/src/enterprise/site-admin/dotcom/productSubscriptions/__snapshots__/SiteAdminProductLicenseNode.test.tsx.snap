// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteAdminProductLicenseNode active 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <SiteAdminProductLicenseNode
      node={
        Object {
          "createdAt": "2020-01-01",
          "id": "l1",
          "info": Object {
            "__typename": "ProductLicenseInfo",
            "expiresAt": "2021-01-01",
            "productNameWithBrand": "NB",
            "tags": Array [
              "a",
            ],
            "userCount": 123,
          },
          "licenseKey": "lk1",
          "subscription": Object {
            "activeLicense": Object {
              "id": "l1",
            },
            "name": "s",
            "urlForSiteAdmin": "/s",
          },
        }
      }
      showSubscription={true}
    >
      <li
        className="list-group-item py-2"
      >
        <div
          className="d-flex align-items-center justify-content-between"
        >
          <div
            className="mr-3 text-truncate"
          >
            <strong>
              License in
               
              <LinkOrSpan
                className="mr-3"
                to="/s"
              >
                <AnchorLink
                  className="mr-3"
                  to="/s"
                >
                  <a
                    className="mr-3"
                    href="/s"
                  >
                    s
                  </a>
                </AnchorLink>
              </LinkOrSpan>
            </strong>
            <span
              className="mr-3"
            >
              <AccountName />
            </span>
          </div>
          <div>
            <ProductLicenseInfoDescription
              className="mr-3"
              licenseInfo={
                Object {
                  "__typename": "ProductLicenseInfo",
                  "expiresAt": "2021-01-01",
                  "productNameWithBrand": "NB",
                  "tags": Array [
                    "a",
                  ],
                  "userCount": 123,
                }
              }
            >
              <span
                className="mr-3"
                title="Tags: a"
              >
                NB
                 (
                123 users
                )
              </span>
            </ProductLicenseInfoDescription>
            <ProductLicenseValidity
              className="d-inline-block mr-3"
              licenseInfo={
                Object {
                  "__typename": "ProductLicenseInfo",
                  "expiresAt": "2021-01-01",
                  "productNameWithBrand": "NB",
                  "tags": Array [
                    "a",
                  ],
                  "userCount": 123,
                }
              }
              primary={false}
            >
              <div
                className="d-inline-block mr-3 "
                data-tooltip="Jan 1, 2021, 12:00:00 AM"
              >
                <strong
                  className="text-success"
                >
                  Valid
                </strong>
                 (
                15 years remaining
                )
              </div>
            </ProductLicenseValidity>
            <span
              className="text-muted"
            >
              Created 
              <Timestamp
                date="2020-01-01"
              >
                <span
                  className="timestamp"
                  data-tooltip="2020-01-01"
                >
                  in almost 14 years
                </span>
              </Timestamp>
            </span>
          </div>
        </div>
        <div>
          Tags: 
          <ProductLicenseTags
            tags={
              Array [
                "a",
              ]
            }
          >
            <div
              className="mr-1 badge badge-secondary"
              key="a"
            >
              a
            </div>
          </ProductLicenseTags>
        </div>
        <CopyableText
          className="mt-2 d-block"
          text="lk1"
        >
          <div
            className="copyable-text form-inline mt-2 d-block"
          >
            <div
              className="input-group"
            >
              <input
                className="copyable-text__input form-control"
                onClick={[Function]}
                readOnly={true}
                type="text"
                value="lk1"
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <Memo(ContentCopyIcon)
                    className="icon-inline"
                  >
                    <svg
                      className="mdi-icon icon-inline"
                      fill="currentColor"
                      height={24}
                      viewBox="0 0 24 24"
                      width={24}
                    >
                      <path
                        d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
                      />
                    </svg>
                  </Memo(ContentCopyIcon)>
                   
                  Copy
                </button>
              </div>
            </div>
          </div>
        </CopyableText>
      </li>
    </SiteAdminProductLicenseNode>
  </Router>
</MemoryRouter>
`;

exports[`SiteAdminProductLicenseNode inactive 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <SiteAdminProductLicenseNode
      node={
        Object {
          "createdAt": "2020-01-01",
          "id": "l1",
          "info": Object {
            "__typename": "ProductLicenseInfo",
            "expiresAt": "2021-01-01",
            "productNameWithBrand": "NB",
            "tags": Array [
              "a",
            ],
            "userCount": 123,
          },
          "licenseKey": "lk1",
          "subscription": Object {
            "activeLicense": Object {
              "id": "l0",
            },
            "name": "s",
            "urlForSiteAdmin": "/s",
          },
        }
      }
      showSubscription={true}
    >
      <li
        className="list-group-item py-2"
      >
        <div
          className="d-flex align-items-center justify-content-between"
        >
          <div
            className="mr-3 text-truncate"
          >
            <strong>
              License in
               
              <LinkOrSpan
                className="mr-3"
                to="/s"
              >
                <AnchorLink
                  className="mr-3"
                  to="/s"
                >
                  <a
                    className="mr-3"
                    href="/s"
                  >
                    s
                  </a>
                </AnchorLink>
              </LinkOrSpan>
            </strong>
            <span
              className="mr-3"
            >
              <AccountName />
            </span>
          </div>
          <div>
            <ProductLicenseInfoDescription
              className="mr-3"
              licenseInfo={
                Object {
                  "__typename": "ProductLicenseInfo",
                  "expiresAt": "2021-01-01",
                  "productNameWithBrand": "NB",
                  "tags": Array [
                    "a",
                  ],
                  "userCount": 123,
                }
              }
            >
              <span
                className="mr-3"
                title="Tags: a"
              >
                NB
                 (
                123 users
                )
              </span>
            </ProductLicenseInfoDescription>
            <span
              className="text-warning font-weight-bold mr-3"
              data-tooltip="A newer license was generated for this subscription. This license should no longer be used."
            >
              Inactive
            </span>
            <span
              className="text-muted"
            >
              Created 
              <Timestamp
                date="2020-01-01"
              >
                <span
                  className="timestamp"
                  data-tooltip="2020-01-01"
                >
                  in almost 14 years
                </span>
              </Timestamp>
            </span>
          </div>
        </div>
        <div>
          Tags: 
          <ProductLicenseTags
            tags={
              Array [
                "a",
              ]
            }
          >
            <div
              className="mr-1 badge badge-secondary"
              key="a"
            >
              a
            </div>
          </ProductLicenseTags>
        </div>
        <CopyableText
          className="mt-2 d-block"
          text="lk1"
        >
          <div
            className="copyable-text form-inline mt-2 d-block"
          >
            <div
              className="input-group"
            >
              <input
                className="copyable-text__input form-control"
                onClick={[Function]}
                readOnly={true}
                type="text"
                value="lk1"
              />
              <div
                className="input-group-append"
              >
                <button
                  className="btn btn-secondary"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <Memo(ContentCopyIcon)
                    className="icon-inline"
                  >
                    <svg
                      className="mdi-icon icon-inline"
                      fill="currentColor"
                      height={24}
                      viewBox="0 0 24 24"
                      width={24}
                    >
                      <path
                        d="M19,21H8V7H19M19,5H8A2,2 0 0,0 6,7V21A2,2 0 0,0 8,23H19A2,2 0 0,0 21,21V7A2,2 0 0,0 19,5M16,1H4A2,2 0 0,0 2,3V17H4V3H16V1Z"
                      />
                    </svg>
                  </Memo(ContentCopyIcon)>
                   
                  Copy
                </button>
              </div>
            </div>
          </div>
        </CopyableText>
      </li>
    </SiteAdminProductLicenseNode>
  </Router>
</MemoryRouter>
`;
